sklearn/multioutput.py

```diff
@@ -0,0 +1,151 @@
+        using `n_jobs>1` can result in slower performance due
+        to the overhead of spawning processes.
+    """
+    def __init__(self, estimator, n_jobs=1):
+        self.estimator = estimator
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y, sample_weight=None):
+        """Fit underlying regressors
+
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+            Data.
+
+        y : array-like, shape = [n_samples, n_targets]
+            The multiple target values.
+
+        sample_weight : array-like, shape = [n_samples] or None
+            Sample weights. If None, then samples are equally weighted.
+            Only supported if the underlying regressor supports sample
+            weights.
+
+        Returns
+        -------
+        self
+        """
+        X, y = check_X_y(X, y,
+                         multi_output=True,
+                         accept_sparse=['csr', 'csc', 'coo', 'dok', 'lil'])
+
+        if y.ndim == 1:
+            raise ValueError("y must have at least two dimensions for "
+                             "multi target regression but has only one.")
+
+        if (sample_weight is not None and
+            not has_fit_parameter(self.estimator, 'sample_weight')):
+            raise ValueError("Underlying regressor does not support sample weights.")
+
+        self.estimators_ = Parallel(n_jobs=self.n_jobs)(delayed(_fit_regression)(
+            self.estimator, X, y[:, i], sample_weight) for i in range(y.shape[1]))
+        return self
+
+    def predict(self, X):
+        """Predict regression values for X.
+
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+            Data.
+
+        Returns
+        -------
+        y : array of shape [n_samples, n_targets]
+            Predicted values.
+        """
+        check_is_fitted(self, 'estimators_')
+
+        X = check_array(X, accept_sparse=['csr', 'csc', 'coo', 'dok', 'lil'])
+
+        pred = Parallel(n_jobs=self.n_jobs)(delayed(parallel_helper)(e, 'predict', X)
+                                            for e in self.estimators_)
+
+        return np.asarray(pred).T
+
+    def score(self, X, y, sample_weight=None):
+        """Returns the coefficient of determination R^2 of the prediction.
+
+        The coefficient R^2 is defined as (1 - u/v), where u is the regression
+        sum of squares ((y_true - y_pred) ** 2).sum() and v is the residual
+        sum of squares ((y_true - y_true.mean()) ** 2).sum().
+        Best possible score is 1.0 and it can be negative (because the
+        model can be arbitrarily worse). A constant model that always
+        predicts the expected value of y, disregarding the input features,
+        would get a R^2 score of 0.0.
+
+        Note
+        ----
+        R^2 is calculated by weighting all the targets equally using
+        multioutput=uniform_average.
```

Comment:
Yes, `RandomForestRegressor` uses the default `score` method where `multioutput='variance_weighted'` which as I understand is going to change to `uniform_average`.

Should I extend the note to make that clearer?